### PR TITLE
feat: add DCAOrder contracts and fetch DCAOrder data

### PR DESCRIPTION
**Pull request by berteotti**
- Adds DCAOrder contracts
- Fetches cancelled and endTime data from contract
- Remove delete contract logic on error

### CODE REVIEW

File: tenderly.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 account_id: ""
 actions:
-  Swapr/swapr:
+  berteotti/project:
```
Comment (Diogomartf) [inline tenderly.yaml]: We don't want this change

```suggestion
  Swapr/swapr:
```

File: actions/watch.ts [hunk 2]
```diff
@@ -24,28 +24,42 @@ export const checkForAndPlaceOrder: ActionFn = async (
...
+        if(cancelledResult.status === "rejected" || endTimeResult.status === "rejected") {
+          throw "Rejected result"
+        }
+
+        return [cancelledResult.value, endTimeResult.value]
+      })
+
+      //Give it a few more 15 minutes so it reproduces the last order
+      const _15_MINUTES = 900
+      const endTimeIsValid = (parseInt(endTime, 10) + _15_MINUTES) * 1000 > new Date().getTime()
```
Comment (Diogomartf) [inline actions/watch.ts]: can you give more details on this?

File: actions/watch.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ export const checkForAndPlaceOrder: ActionFn = async (
   const registry = await Registry.load(context, blockEvent.network);
   const chainContext = await ChainContext.create(context, blockEvent.network);
 
-  const contractsToDelete = []
+  const contractsToDelete: string[] = []
```
Comment (Diogomartf) [inline actions/watch.ts]: Where are we deleting the contracts addresses?

It's seems we never add to this array.

File: lib/BokkyPooBahsDateTimeLibrary/test/testchain/keystore/UTC--2017-05-20T02-37-30.360937280Z--a00af22d07c87d96eeeb0ed583f8f6ac7812827e
Comment (Diogomartf) [inline lib/BokkyPooBahsDateTimeLibrary/test/testchain/keystore/UTC--2017-05-20T02-37-30.360937280Z--a00af22d07c87d96eeeb0ed583f8f6ac7812827e:1]: do we need this `testchain/keystore`?

File: lib/openzeppelin-contracts/.github/ISSUE_TEMPLATE/bug_report.md
Comment (Diogomartf) [inline lib/openzeppelin-contracts/.github/ISSUE_TEMPLATE/bug_report.md:1]: I think we can delete the .`github` folder

File: lib/openzeppelin-contracts/audit/2017-03.md
Comment (Diogomartf) [inline lib/openzeppelin-contracts/audit/2017-03.md:1]: also think we can delete audits, etc

**Comment by Diogomartf** [comment]
I think we're adding too many things not necessary to run contracts code:

Maybe we can import openzeppelin contracts from this package:
https://github.com/OpenZeppelin/openzeppelin-contracts

Also .`github` folder seems unnecessary, certora, etc

File: actions/watch.ts (hunk 2 above)
Comment (berteotti) [reply actions/watch.ts]: What kind of detail? This is to check if endTime is still valid.

File: actions/watch.ts (hunk 3 above)
Comment (berteotti) [reply actions/watch.ts]: Better to see if everything works fine before we start deleting contracts again as last time deleted unwanted contracts.

File: tenderly.yaml (hunk 1 above)
Comment (berteotti) [reply tenderly.yaml]: fixed

File: actions/watch.ts (hunk 2 above)
Comment (berteotti) [reply actions/watch.ts]: Gave it a diff name

File: actions/watch.ts (hunk 3 above)
Comment (Diogomartf) [reply actions/watch.ts]: Yeah, but leaving it here doesn't make sense then. We should add it when we add the functionality.

**Review by Diogomartf — APPROVED** [review]

**Review by Diogomartf — APPROVED** [review]

**Pull request merged**